txpool/pool.go

```diff
@@ -2187,19 +2193,28 @@ func (mt *metaTx) better(than *metaTx, pendingBaseFee uint64) bool {
 
 	switch mt.currentSubPool {
 	case PendingSubPool:
-		var effectiveTip, thanEffectiveTip uint64
-		if pendingBaseFee <= mt.minFeeCap {
-			effectiveTip = cmp.Min(mt.minFeeCap-pendingBaseFee, mt.minTip)
+		var effectiveTip, thanEffectiveTip uint256.Int
+		if mt.minFeeCap.Cmp(&pendingBaseFee) >= 0 {
+			if mt.minFeeCap.Cmp(uint256.NewInt(mt.minTip)) <= 0 {
+				effectiveTip = mt.minFeeCap
+			} else {
+				effectiveTip = *uint256.NewInt(mt.minTip)
+			}
 		}
-		if pendingBaseFee <= than.minFeeCap {
-			thanEffectiveTip = cmp.Min(than.minFeeCap-pendingBaseFee, than.minTip)
+		if than.minFeeCap.Cmp(&pendingBaseFee) >= 0 {
+			than.minFeeCap.Sub(&than.minFeeCap, &pendingBaseFee)
```

Comment:
Don't change `than.minFeeCap`. Introduce a temporary variable for the difference instead.